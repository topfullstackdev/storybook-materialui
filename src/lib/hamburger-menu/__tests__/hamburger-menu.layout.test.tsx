import { render } from '@testing-library/react';

import { HamburgerMenu } from '../hamburger-menu';

describe('HamburgerMenu', () => {
  it('should render successfully', () => {
    function HamburgerMenuWrapper() {
      return (
        <HamburgerMenu
          items={[
            {
              label: 'My Dashboard',
              onClick: () => 'My Dashboard',
            },
            {
              label: 'Profile',
              onClick: () => 'Profile',
            },
          ]}
        />
      );
    }

    const { container } = render(<HamburgerMenuWrapper />);
    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it('should be in hamburger icon', () => {
    function HamburgerMenuWrapper() {
      return (
        <HamburgerMenu
          items={[
            {
              label: 'My Dashboard',
              onClick: () => 'My Dashboard',
            },
            {
              label: 'Profile',
              onClick: () => 'Profile',
            },
          ]}
        />
      );
    }

    const { container, getByRole } = render(<HamburgerMenuWrapper />);
    const icon = getByRole('button');
    expect(container).toMatchSnapshot();
    expect(icon).toBeTruthy();
  });

  it('should be in menu', () => {
    function HamburgerMenuWrapper() {
      return (
        <HamburgerMenu
          items={[
            {
              label: 'My Dashboard',
              onClick: () => 'My Dashboard',
            },
            {
              label: 'Profile',
              onClick: () => 'Profile',
            },
          ]}
        />
      );
    }

    const { container, getByText } = render(<HamburgerMenuWrapper />);
    const element = getByText('My Dashboard');
    expect(container).toMatchSnapshot();
    expect(element).toBeInTheDocument();
  });
});
