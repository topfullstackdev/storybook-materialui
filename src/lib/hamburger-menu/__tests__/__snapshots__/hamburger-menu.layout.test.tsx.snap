// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu should be in hamburger icon 1`] = `
.emotion-1 {
  display: block;
  position: relative;
}

.emotion-1 .ReactCollapse--collapse {
  -webkit-transition: height 400ms;
  transition: height 400ms;
  position: absolute;
  top: 40px;
  left: 2px;
  z-index: 99;
  width: 216px;
  padding-left: 1px;
}

.emotion-3 {
  background: 0 0;
  display: block;
  position: relative;
  float: left;
  overflow: hidden;
  margin: 0;
  padding: 0;
  width: 34px;
  height: 27px;
  font-size: 0;
  text-indent: -9999px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  box-shadow: none;
  border-radius: none;
  border: none;
  cursor: pointer;
  outline: none;
}

.emotion-3 span {
  display: block;
  position: absolute;
  top: 10px;
  left: 0;
  right: 0;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
}

.emotion-3 span::after {
  content: '';
  position: absolute;
  display: block;
  left: 0;
  width: 100%;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
  transform-origin: bottom right;
  bottom: -10px;
}

.emotion-3 span::before {
  content: '';
  position: absolute;
  display: block;
  left: 0;
  width: 100%;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
  transform-origin: top right;
  top: -10px;
}

.emotion-5 {
  width: 215px;
  box-shadow: 0 2px 0.5px 0.7px rgb(51 51 51 / 35%);
  background-color: white;
  z-index: 99;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-bottom: 1px solid #313131;
  font-size: 14px;
  color: #333;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-7.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-7.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-7.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-7.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-7.Mui-disabled {
  opacity: 0.38;
}

.emotion-7+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-7+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-7 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-7 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-7 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-7 {
    min-height: auto;
  }
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  font-weight: normal;
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Verdana,Arial,sans-serif;
}

.emotion-10 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-bottom: none;
  font-size: 14px;
  color: #333;
}

.emotion-11::-moz-focus-inner {
  border-style: none;
}

.emotion-11.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-11 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-11:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-11:hover {
    background-color: transparent;
  }
}

.emotion-11.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-11.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-11.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-11.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-11.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-11.Mui-disabled {
  opacity: 0.38;
}

.emotion-11+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-11+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-11 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-11 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-11 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-11 {
    min-height: auto;
  }
}

<div>
  <div
    class="emotion-0 MuiBox-root emotion-1"
  >
    <button
      aria-label="Open Navigation"
      class="emotion-2 MuiBox-root emotion-3"
      data-cy="Hamburger"
    >
      <span
        class="MuiBox-root emotion-4"
      />
    </button>
    <div
      aria-hidden="true"
      class="ReactCollapse--collapse"
      style="height: 0px; overflow: hidden;"
    >
      <div
        class="ReactCollapse--content"
      >
        <nav
          class="emotion-5 emotion-6"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root emotion-7"
            data-cy="hamburger-menu-item:My Dashboard"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-8 emotion-9"
            >
              My Dashboard
            </p>
            <span
              class="MuiTouchRipple-root emotion-10"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root emotion-11"
            data-cy="hamburger-menu-item:Profile"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-8 emotion-9"
            >
              Profile
            </p>
            <span
              class="MuiTouchRipple-root emotion-10"
            />
          </li>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`HamburgerMenu should be in menu 1`] = `
.emotion-1 {
  display: block;
  position: relative;
}

.emotion-1 .ReactCollapse--collapse {
  -webkit-transition: height 400ms;
  transition: height 400ms;
  position: absolute;
  top: 40px;
  left: 2px;
  z-index: 99;
  width: 216px;
  padding-left: 1px;
}

.emotion-3 {
  background: 0 0;
  display: block;
  position: relative;
  float: left;
  overflow: hidden;
  margin: 0;
  padding: 0;
  width: 34px;
  height: 27px;
  font-size: 0;
  text-indent: -9999px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  box-shadow: none;
  border-radius: none;
  border: none;
  cursor: pointer;
  outline: none;
}

.emotion-3 span {
  display: block;
  position: absolute;
  top: 10px;
  left: 0;
  right: 0;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
}

.emotion-3 span::after {
  content: '';
  position: absolute;
  display: block;
  left: 0;
  width: 100%;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
  transform-origin: bottom right;
  bottom: -10px;
}

.emotion-3 span::before {
  content: '';
  position: absolute;
  display: block;
  left: 0;
  width: 100%;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
  transform-origin: top right;
  top: -10px;
}

.emotion-5 {
  width: 215px;
  box-shadow: 0 2px 0.5px 0.7px rgb(51 51 51 / 35%);
  background-color: white;
  z-index: 99;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-bottom: 1px solid #313131;
  font-size: 14px;
  color: #333;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-7.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-7.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-7.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-7.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-7.Mui-disabled {
  opacity: 0.38;
}

.emotion-7+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-7+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-7 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-7 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-7 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-7 {
    min-height: auto;
  }
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  font-weight: normal;
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Verdana,Arial,sans-serif;
}

.emotion-10 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-bottom: none;
  font-size: 14px;
  color: #333;
}

.emotion-11::-moz-focus-inner {
  border-style: none;
}

.emotion-11.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-11 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-11:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-11:hover {
    background-color: transparent;
  }
}

.emotion-11.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-11.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-11.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-11.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-11.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-11.Mui-disabled {
  opacity: 0.38;
}

.emotion-11+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-11+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-11 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-11 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-11 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-11 {
    min-height: auto;
  }
}

<div>
  <div
    class="emotion-0 MuiBox-root emotion-1"
  >
    <button
      aria-label="Open Navigation"
      class="emotion-2 MuiBox-root emotion-3"
      data-cy="Hamburger"
    >
      <span
        class="MuiBox-root emotion-4"
      />
    </button>
    <div
      aria-hidden="true"
      class="ReactCollapse--collapse"
      style="height: 0px; overflow: hidden;"
    >
      <div
        class="ReactCollapse--content"
      >
        <nav
          class="emotion-5 emotion-6"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root emotion-7"
            data-cy="hamburger-menu-item:My Dashboard"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-8 emotion-9"
            >
              My Dashboard
            </p>
            <span
              class="MuiTouchRipple-root emotion-10"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root emotion-11"
            data-cy="hamburger-menu-item:Profile"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-8 emotion-9"
            >
              Profile
            </p>
            <span
              class="MuiTouchRipple-root emotion-10"
            />
          </li>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`HamburgerMenu should render successfully 1`] = `
.emotion-1 {
  display: block;
  position: relative;
}

.emotion-1 .ReactCollapse--collapse {
  -webkit-transition: height 400ms;
  transition: height 400ms;
  position: absolute;
  top: 40px;
  left: 2px;
  z-index: 99;
  width: 216px;
  padding-left: 1px;
}

.emotion-3 {
  background: 0 0;
  display: block;
  position: relative;
  float: left;
  overflow: hidden;
  margin: 0;
  padding: 0;
  width: 34px;
  height: 27px;
  font-size: 0;
  text-indent: -9999px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  box-shadow: none;
  border-radius: none;
  border: none;
  cursor: pointer;
  outline: none;
}

.emotion-3 span {
  display: block;
  position: absolute;
  top: 10px;
  left: 0;
  right: 0;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
}

.emotion-3 span::after {
  content: '';
  position: absolute;
  display: block;
  left: 0;
  width: 100%;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
  transform-origin: bottom right;
  bottom: -10px;
}

.emotion-3 span::before {
  content: '';
  position: absolute;
  display: block;
  left: 0;
  width: 100%;
  height: 4px;
  background-color: #1976d2;
  border-radius: 3px;
  transform-origin: top right;
  top: -10px;
}

.emotion-5 {
  width: 215px;
  box-shadow: 0 2px 0.5px 0.7px rgb(51 51 51 / 35%);
  background-color: white;
  z-index: 99;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-bottom: 1px solid #313131;
  font-size: 14px;
  color: #333;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-7.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-7.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-7.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-7.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-7.Mui-disabled {
  opacity: 0.38;
}

.emotion-7+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-7+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-7 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-7 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-7 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-7 {
    min-height: auto;
  }
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  font-weight: normal;
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Verdana,Arial,sans-serif;
}

.emotion-10 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-bottom: none;
  font-size: 14px;
  color: #333;
}

.emotion-11::-moz-focus-inner {
  border-style: none;
}

.emotion-11.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-11 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-11:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-11:hover {
    background-color: transparent;
  }
}

.emotion-11.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-11.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-11.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-11.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-11.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-11.Mui-disabled {
  opacity: 0.38;
}

.emotion-11+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-11+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-11 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-11 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-11 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-11 {
    min-height: auto;
  }
}

<div>
  <div
    class="emotion-0 MuiBox-root emotion-1"
  >
    <button
      aria-label="Open Navigation"
      class="emotion-2 MuiBox-root emotion-3"
      data-cy="Hamburger"
    >
      <span
        class="MuiBox-root emotion-4"
      />
    </button>
    <div
      aria-hidden="true"
      class="ReactCollapse--collapse"
      style="height: 0px; overflow: hidden;"
    >
      <div
        class="ReactCollapse--content"
      >
        <nav
          class="emotion-5 emotion-6"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root emotion-7"
            data-cy="hamburger-menu-item:My Dashboard"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-8 emotion-9"
            >
              My Dashboard
            </p>
            <span
              class="MuiTouchRipple-root emotion-10"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root emotion-11"
            data-cy="hamburger-menu-item:Profile"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-8 emotion-9"
            >
              Profile
            </p>
            <span
              class="MuiTouchRipple-root emotion-10"
            />
          </li>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
